pyproject.toml

```diff
@@ -12,16 +12,157 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "nemo_curator"
+description = "Scalable Data Preprocessing Tool for Training Large Language Models"
+readme = { file = "README.md", content-type = "text/markdown" }
+authors = [
+    { name = "Joseph Jennings", email = "jjennings@nvidia.com" },
+    { name = "Mostofa Patwary", email = "mpatwary@nvidia.com" },
+    { name = "Sandeep Subramanian", email = "sasubramania@nvidia.com" },
+    { name = "Shrimai Prabhumoye", email = "sprabhumoye@nvidia.com" },
+    { name = "Ayush Dattagupta", email = "adattagupta@nvidia.com" },
+    { name = "Vibhu Jawa", email = "vjawa@nvidia.com" },
+    { name = "Jiwei Liu", email = "jiweil@nvidia.com" },
+    { name = "Ryan Wolf", email = "rywolf@nvidia.com" },
+    { name = "Sarah Yurick", email = "syurick@nvidia.com" },
+]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+]
+requires-python = ">=3.10, <3.11"
+dependencies = [
+    "awscli>=1.22.55",
+    "beautifulsoup4",
+    "charset_normalizer>=3.1.0",
+    "comment_parser",
+    "crossfit>=0.0.6",
+    "dask-mpi>=2021.11.0",
+    "dask[complete]>=2021.7.1",
+    "distributed>=2021.7.1",
+    "fasttext==0.9.2",
+    "ftfy==6.1.1",
+    "in-place==0.5.0",
+    "jieba==0.42.1",
+    "justext==3.0.1",
+    "lxml_html_clean",
+    "mwparserfromhell==0.6.5",
+    "nemo_toolkit[nlp]>=1.23.0",
+    "numpy<2",
+    "openai",
+    "peft",
+    "presidio-analyzer==2.2.351",
+    "presidio-anonymizer==2.2.351",
+    "pycld2",
+    "resiliparse",
+    "spacy>=3.6.0, <3.8.0",
+    "unidic-lite==1.0.8",
+    "usaddress==0.5.10",
+    "warcio==1.7.4",
+    "zstandard==0.18.0",
+]
+dynamic = ["version"]
+
+[project.optional-dependencies]
```

Comment:
Curious for opinions on this new implementation. Instead of having the `RAPIDS_NIGHTLY` flag, the user installs `.[image_nightly]` or `.[all_nightly`.

Reply:
I definitely think that optional dependencies are a cleaner solution for this.